ipfs/defaults.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/ipfs/interface-go-ipfs-core/options"
+)
+
+// BootstrapPeers is a list of default bootstrap peers for IPFS network
+// TODO: Change this to LL defaults at some point
+var BootstrapPeers, _ = ipfscfg.DefaultBootstrapPeers()
+
+// DefaultFullNodeConfig provides default embedded IPFS configuration for FullNode
+func DefaultFullNodeConfig() (*ipfscfg.Config, error) {
+	identity, err := ipfscfg.CreateIdentity(os.Stdout, []options.KeyGenerateOption{
+		options.Key.Type(options.Ed25519Key),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var conf = &ipfscfg.Config{
+		// ed25519 PKI identity for p2p communication
+		// TODO: Any reason not reuse ed25519 key from consensus identity?
+		Identity: identity,
+		// List of libp2p peer addresses the node automatically connects on start
+		Bootstrap: ipfscfg.BootstrapPeerStrings(BootstrapPeers),
+		// Configure all the addresses the node listens to
+		Addresses: ipfscfg.Addresses{
+			// List of libp2p addresses to listen to
+			Swarm: []string{
+				// listen on all network interfaces.
+				// NOTE: We don't use IPv6 for now, but we might add support for it at some point.
+				"/ip4/0.0.0.0/tcp/4001",
+			},
+			API: []string{
+				"/ip4/127.0.0.1/tcp/5001",
+			},
+			// List of public libp2p WAN address the Node can be connected with
+			// NOTE: Should be configured by Node administrator manually
+			Announce: nil,
+			// List of libp2p address the Node listens to but not accessible from the WAN.
+			// It's a good manner to fil this, as that preserves connected peers from trying connections to those
+			// NOTE: Should be configured by Node administrator manually
+			NoAnnounce: nil,
+			// List of addresses IPFS API server listens to
+			// we do not run IPFS Gateway
+			Gateway: nil,
+		},
+		// Networking configuration
+		Swarm: ipfscfg.SwarmConfig{
+			// auto port forwarding is nice
+			DisableNatPortMap: false,
+			// metrics are useful
+			DisableBandwidthMetrics: false,
+			// relays are cool from engineering standpoint, but don't use them
+			EnableRelayHop: false, EnableAutoRelay: false,
+			// ConnMgr manager controls the amount of conns a Node holds. If the amount of conns hits HighWater, ConnMgr
+			// trims amount of connections down to LowWater
+			// NOTE: Connections can by prioritized using Tagging in ConnMgr in order to keep important conns alive
+			ConnMgr: ipfscfg.ConnMgr{
+				// we use defaults from IPFS for HighWater
+				HighWater: 900,
+				// we use defaults from IPFS for HighWater
+				LowWater: 600,
+				// defines how often ConnMgr should check conns for trimming, we use default from IPFS
+				GracePeriod: "20s",
+				// just a basic connection manager
+				Type: "basic",
+			},
+			// Configuration for all networking protocols
+			Transports: ipfscfg.Transports{
+				Network: transport{
+					// we default to TCP as it's most common, battle tested transport, but we may migrate to the one below
+					TCP: ipfscfg.False,
```

Comment:
Then, shouldn't this be:
```suggestion
					TCP: ipfscfg.True,
```

Reply:
Eagle Eye!